File: implementations/typescript/packages/command/test/command.test.js
import assert from "node:assert/strict";
import test from "node:test";
import { COMMAND_EVENT_TYPES, isCommandEventType } from "../src/index.js";

test("command event types includes all 5 registry entries", () => {
  const expected = [
    "command.requested",
    "command.accepted",
    "command.rejected",
    "command.completed",
    "command.failed"
  ];
  for (const type of expected) {
    assert.equal(COMMAND_EVENT_TYPES.has(type), true, `missing: ${type}`);
  }
  assert.equal(COMMAND_EVENT_TYPES.size, expected.length);
});

test("isCommandEventType positives", () => {
  assert.equal(isCommandEventType("command.requested"), true);
  assert.equal(isCommandEventType("command.accepted"), true);
  assert.equal(isCommandEventType("command.rejected"), true);
  assert.equal(isCommandEventType("command.completed"), true);
  assert.equal(isCommandEventType("command.failed"), true);
});

test("isCommandEventType negatives", () => {
  assert.equal(isCommandEventType("task.submitted"), false);
  assert.equal(isCommandEventType("session.opened"), false);
  assert.equal(isCommandEventType("query.requested"), false);
  assert.equal(isCommandEventType(""), false);
});

File: implementations/go/command/command_test.go
package command

import "testing"

func TestCommandEventTypesIncludesAllEntries(t *testing.T) {
	expected := []string{
		"command.requested",
		"command.accepted",
		"command.rejected",
		"command.completed",
		"command.failed",
	}
	for _, typ := range expected {
		if !EventTypes[typ] {
			t.Errorf("missing: %s", typ)
		}
	}
	if len(EventTypes) != len(expected) {
		t.Errorf("size mismatch: got %d, want %d", len(EventTypes), len(expected))
	}
}

func TestIsEventTypePositives(t *testing.T) {
	tests := []string{
		"command.requested",
		"command.accepted",
		"command.rejected",
		"command.completed",
		"command.failed",
	}
	for _, typ := range tests {
		if !IsEventType(typ) {
			t.Errorf("expected true for %s", typ)
		}
	}
}

func TestIsEventTypeNegatives(t *testing.T) {
	tests := []string{
		"task.submitted",
		"session.opened",
		"query.requested",
		"",
	}
	for _, typ := range tests {
		if IsEventType(typ) {
			t.Errorf("expected false for %s", typ)
		}
	}
}
